featureflags.go

```diff
@@ -107,11 +108,17 @@ func (poller *FeatureFlagsPoller) fetchNewFeatureFlags() {
 		poller.Errorf("Unable to unmarshal response from api/feature_flag", err)
 		return
 	}
-	poller.mutex.Lock()
 	if !poller.fetchedFlagsSuccessfullyOnce {
 		poller.loaded <- true
 	}
-	poller.featureFlags = featureFlagsResponse.Results
+	newFlags := []FeatureFlag{}
+	for _, flag := range featureFlagsResponse.Results {
+		if flag.Active {
+			newFlags = append(newFlags, flag)
+		}
+	}
+	poller.mutex.Lock()
+	poller.featureFlags = newFlags
 	poller.mutex.Unlock()
```

Comment:
👍

Reply:
issue wasn't the mutex btw - was just that we weren't filtering for active flags (in any lib 😬 )

Reply:
no no, I know mutex wasn't issue. This is just more clean locking on just where the mutation was happening.

>(in any lib 😬 )
🥶

Reply:
yeah indeed